Build Failed || java.lang.IllegalStateException at DefaultCacheAwareContextLoaderDelegate.java

java.lang.IllegalStateException at DefaultCacheAwareContextLoaderDelegate.java:132
	Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:800
	Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException at DefaultListableBeanFactory.java:1799

--- a/backend/src/test/kotlin/com/smartcity/education/backend/controllers/AssessmentApiControllerTests.kt
+++ b/backend/src/test/kotlin/com/smartcity/education/backend/controllers/AssessmentApiControllerTests.kt
@@ -4,6 +4,7 @@ import com.smartcity.education.backend.assigners.AssessmentAssigner
 import com.smartcity.education.backend.authentication.AuthUtil
 import com.smartcity.education.backend.models.*
 import com.smartcity.education.backend.repositories.AssessmentRepository
+import com.smartcity.education.backend.repositories.MatriculationRepository
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
 import org.mockito.Mockito
@@ -20,6 +21,8 @@ import java.util.*
 class AssessmentApiControllerTests {
     @MockBean
     private val repository: AssessmentRepository? = null
+    @MockBean
+    private val matriculationRepository: MatriculationRepository? = null
     @MockBean
     private val assigner: AssessmentAssigner? = null
     @MockBean
@@ -105,6 +108,8 @@ class AssessmentApiControllerTests {
                 grade = 1.0f,
                 date = LocalDateTime.now()
         )
+        val matriculationId = 1L
+        grade.matriculationId = matriculationId
         val obj = Assessment(
                 title = ""
         )
@@ -116,6 +121,7 @@ class AssessmentApiControllerTests {
         Assertions.assertEquals(result?.statusCode, HttpStatus.CREATED)
 
         verify(repository, times(1))?.findById(id)
+        verify(matriculationRepository, times(1))?.findById(matriculationId)
         verify(authUtil, times(1))?.hasInstitutionAuthority(SecurityContextHolder.getContext(), -1)
         verify(repository, times(1))?.save(obj)
     }

--- a/backend/src/test/kotlin/com/smartcity/education/backend/controllers/MatriculationApiControllerTests.kt
+++ b/backend/src/test/kotlin/com/smartcity/education/backend/controllers/MatriculationApiControllerTests.kt
@@ -3,6 +3,7 @@ package com.smartcity.education.backend.controllers
 import com.smartcity.education.backend.assigners.MatriculationAssigner
 import com.smartcity.education.backend.authentication.AuthUtil
 import com.smartcity.education.backend.models.*
+import com.smartcity.education.backend.repositories.AssessmentRepository
 import com.smartcity.education.backend.repositories.MatriculationRepository
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
@@ -19,6 +20,8 @@ import java.util.*
 class MatriculationApiControllerTests {
     @MockBean
     private val repository: MatriculationRepository? = null
+    @MockBean
+    private val assessmentRepository: AssessmentRepository? = null
     @MockBean
     private val assigner: MatriculationAssigner? = null
     @MockBean
@@ -138,6 +141,8 @@ class MatriculationApiControllerTests {
                 grade = 1.0f,
                 date = LocalDateTime.now()
         )
+        val assessmentId = 1L
+        grade.assessmentId = assessmentId
         val obj = Matriculation(
                 date = LocalDateTime.now()
         )
@@ -149,6 +154,7 @@ class MatriculationApiControllerTests {
         Assertions.assertEquals(result?.statusCode, HttpStatus.CREATED)
 
         verify(repository, times(1))?.findById(id)
+        verify(assessmentRepository, times(1))?.findById(assessmentId)
         verify(authUtil, times(1))?.hasInstitutionAuthority(SecurityContextHolder.getContext(), -1)
         verify(repository, times(1))?.save(obj)
     }

